eslint-plugin-prettier.js

```diff
@@ -0,0 +1,306 @@
+ * @returns {void}
+ */
+function reportDelete(context, offset, text) {
+  const start = getLocFromIndex(context, offset);
+  const end = getLocFromIndex(context, offset + text.length);
+  const range = [offset, offset + text.length];
+  context.report({
+    message: 'Delete `{{ code }}`',
+    data: { code: showInvisibles(text) },
+    loc: { start, end },
+    fix(fixer) {
+      return fixer.removeRange(range);
+    }
+  });
+}
+
+/**
+ * Reports a "Replace ... with ..." issue where text must be replaced.
+ * @param {RuleContext} context - The ESLint rule context.
+ * @param {number} offset - The source offset where to replace deleted text
+ with inserted text.
+ * @param {string} deleteText - The text to be deleted.
+ * @param {string} insertText - The text to be inserted.
+ * @returns {void}
+ */
+function reportReplace(context, offset, deleteText, insertText) {
+  const start = getLocFromIndex(context, offset);
+  const end = getLocFromIndex(context, offset + deleteText.length);
+  const range = [offset, offset + deleteText.length];
+  context.report({
+    message: 'Replace `{{ deleteCode }}` with `{{ insertCode }}`',
+    data: {
+      deleteCode: showInvisibles(deleteText),
+      insertCode: showInvisibles(insertText)
+    },
+    loc: { start, end },
+    fix(fixer) {
+      return fixer.replaceTextRange(range, insertText);
+    }
+  });
+}
+
+// ------------------------------------------------------------------------------
+//  Module Definition
+// ------------------------------------------------------------------------------
+
+module.exports.rules = {
+  prettier: {
+    meta: {
+      fixable: 'code',
+      schema: [
+        // Prettier options:
+        {
+          anyOf: [
+            { enum: [null, 'fb'] },
+            { type: 'object', properties: {}, additionalProperties: true }
+          ]
+        },
+        // Pragma:
+        { type: 'string', pattern: '^@\\w+$' }
+      ]
+    },
+    create(context) {
+      const prettierOptions = context.options[0] === 'fb'
+        ? FB_PRETTIER_OPTIONS
+        : context.options[0];
+
+      const pragma = context.options[1]
+        ? context.options[1].slice(1) // Remove leading @
+        : null;
+
+      if (pragma) {
+        // The pragma is only valid if it is found in a block comment at the
+        // very start of the file.
+        const firstComment = context.getAllComments()[0];
+        if (
+          !(firstComment &&
+            firstComment.type === 'Block' &&
+            firstComment.start === 0)
+        ) {
+          return {};
+        }
+        const parsed = docblock.parse(firstComment.value);
+        if (parsed[pragma] !== '') {
+          return {};
+        }
+      }
+
+      return {
+        Program() {
+          if (!prettier) {
+            // Prettier is expensive to load, so only load it if needed.
+            prettier = require('prettier');
+          }
+          const source = context.getSource();
```

Comment:
I think `context.getSource()` is deprecated -- there doesn't seem to be any documentation for it, and it's in the list of deprecated passthroughs [here](https://github.com/eslint/eslint/blob/0ef09ea071f7df0cfb3f76ddea9ad746cb619912/lib/rule-context.js#L24-L41). Can you replace it with `context.getSourceCode().text` instead?